### PR TITLE
Design docs + roadmap

### PR DESCRIPTION
**Pull request by helfer**
Outlines a roadmap and possible designs for upcoming features. Pull requests for those features are very welcome!

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+* Koa integration
+* Connect integration
+* Better GraphQL error handling
+* Support for simple query timeouts
+* Websocket transport
+
+### Future
+
+* Support for @defer, @stream and @live directives
```
Comment (stubailo) [inline ROADMAP.md]: GraphQL Subscriptions?

Comment (stubailo) [inline ROADMAP.md]: My favorite thing about that is that I believe it can be done without changes to GraphQL-JS.

Comment (helfer) [inline ROADMAP.md]: True, I should add that while I'm at it.

Comment (helfer) [inline ROADMAP.md]: Give me 5 minutes.

**Pull request merged**